### PR TITLE
Replace cofibration syntax with interval expressions

### PR DESCRIPTION
**Pull request by dark-flames**
TODO:

- [x] Modify concrete and parser.
- [x] Modify the type check for PartialEl and Path.
- [x] Move `generic.Cube` to `core` as it is no longer used in `concrete`

### CODE REVIEW

File: base/src/main/java/org/aya/tyck/ExprTycker.java [hunk 1]
```diff
@@ -225,17 +225,13 @@ yield new TermResult(new IntroTerm.Tuple(items.map(Result::wellTyped)),
 
         yield new TermResult(new PrimTerm.Str(litStr.string()), state.primFactory().getCall(PrimDef.ID.STR));
       }
-      case Expr.PartTy par -> {
-        var ty = synthesize(par.type());
-        yield new TermResult(new FormTerm.PartTy(ty.wellTyped(), restr(par.restr())), ty.type());
-      }
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/ExprTycker.java:231]: Any rationale?

File: base/src/main/java/org/aya/tyck/ExprTycker.java [hunk 2]
```diff
@@ -287,14 +278,15 @@ private boolean boundary(@NotNull Expr loc, @NotNull Term lhs, @NotNull Term rhs
     return happy;
   }
 
-  private @NotNull Option<Restr.Side<Term>> clause(@NotNull Restr.Side<Expr> clause, @NotNull Term type) {
-    var cofib = new Restr.Cofib<>(clause.cof().ands().map(this::condition));
+  private @NotNull Option<Restr.Side<Term>> clause(@NotNull Expr lhs, @NotNull Expr rhs, @NotNull Term rhsType) {
+    var intervalTerm = inherit(lhs, PrimTerm.Interval.INSTANCE).wellTyped();
+    var cofib = new Restr.Cofib<>(ImmutableSeq.of(new Restr.Cond<>(intervalTerm, true)));
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/ExprTycker.java:283]: You should _only_ have `RefTerm` in the first argument of `Cond`

File: base/src/main/java/org/aya/tyck/ExprTycker.java (hunk 1 above)
Comment (ice1000) [reply base/src/main/java/org/aya/tyck/ExprTycker.java:231]: I was stupid

**Review by imkiva — APPROVED** [review]
I just can't wait to see it land!! so

bors merge

**Pull request merged**